src/IssueViz/IssueVisualizationControl/ViewModels/IssueVisualizationViewModel.cs

```diff
@@ -0,0 +1,207 @@
+                    OnPropertyChanged(nameof(CurrentLocationListItem));
+                }
+                else
+                {
+                    selectionEvents.SelectedLocation = currentLocation?.Location;
+                }
+            }
+        }
+
+        private void SelectionEvents_SelectedIssueChanged(object sender, IssueChangedEventArgs e)
+        {
+            CurrentIssue = e.Issue;
+        }
+
+        private void SelectionEventsOnSelectedFlowChanged(object sender, FlowChangedEventArgs e)
+        {
+            isBindingUpdatedOutsideOfControl = true;
+
+            CurrentFlow = e.Flow;
+
+            isBindingUpdatedOutsideOfControl = false;
+        }
+
+        private void SelectionEvents_SelectedLocationChanged(object sender, LocationChangedEventArgs e)
+        {
+            isBindingUpdatedOutsideOfControl = true;
+
+            if (e.Location == null)
+            {
+                CurrentLocationListItem = null;
+            }
+            else
+            {
+                var selectedLocationListItem = LocationListItems?
+                    .OfType<LocationListItem>()
+                    .FirstOrDefault(x => x.Location == e.Location);
+
+                CurrentLocationListItem = selectedLocationListItem;
+            }
+
+            isBindingUpdatedOutsideOfControl = false;
+        }
+
+        private IReadOnlyList<ILocationListItem> BuildLocationListItems(IAnalysisIssueFlowVisualization flow)
+        {
+            var listItems = new List<ILocationListItem>();
+            var flowLocations = flow?.Locations.ToList();
+
+            if (flowLocations != null && flowLocations.Any())
+            {
+                for (var i = 0; i < flowLocations.Count; i++)
+                {
+                    var filePath = flowLocations[i].Location.FilePath;
+                    var fileIcon = GetImageMonikerForFile(filePath);
+                    listItems.Add(new FileNameLocationListItem(filePath, Path.GetFileName(filePath), fileIcon));
+
+                    var sequentialLocations = flowLocations.Skip(i).TakeWhile(x => x.Location.FilePath == filePath).ToList();
+                    listItems.AddRange(sequentialLocations.Select(x => (ILocationListItem)new LocationListItem(x)));
+
+                    i += sequentialLocations.Count - 1;
+                }
+            }
+
+            return listItems;
+        }
+
+        private ImageMoniker GetImageMonikerForFile(string filePath)
+        {
+            try
+            {
+                return vsImageService.GetImageMonikerForFile(filePath);
+            }
+            catch (Exception e) when (!ErrorHandler.IsCriticalException(e))
+            {
+                logger.WriteLine(Resources.ERR_FailedToGetFileImageMoniker, filePath, e);
+
+                return KnownMonikers.Blank;
+            }
+        }
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        private void OnPropertyChanged(string propertyName)
```

Comment:
It's a common pattern to use `[CallerMemberName]` here to avoid specifying the property name in the default case.

Reply:
Hm, it's not what I want though - I want to pass null to completely render the UI (see the [remarks](https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.inotifypropertychanged.propertychanged?redirectedfrom=MSDN&view=netcore-3.1))